src/docfx/config/ops/OpsConfigAdapter.cs

```diff
@@ -47,18 +78,100 @@ public static JObject Load(FileResolver fileResolver, SourceInfo<string> name, s
                 {
                     ["defaultLocale"] = GetDefaultLocale(docset.site_name),
                 },
+                ["monikerDefinition"] = MonikerDefinitionApi,
+                ["markdownValidationRules"] = $"{MarkdownValidationRulesApi}{metadataServiceQueryParams}",
+                ["metadataSchema"] = new JArray(
+                    Path.Combine(AppContext.BaseDirectory, "data/schemas/OpsMetadata.json"),
+                    $"{MetadataSchemaApi}{metadataServiceQueryParams}"),
             };
         }
 
-        private static string ResolveFile(FileResolver fileResolver, SourceInfo<string> name, string url)
+        public static async Task<HttpResponseMessage> InterceptHttpRequest(ErrorLog errorLog, HttpRequestMessage request)
+        {
+            foreach (var (baseUrl, rule) in s_apis)
+            {
+                if (request.RequestUri.OriginalString.StartsWith(baseUrl))
+                {
+                    return new HttpResponseMessage { Content = new StringContent(await rule(errorLog, request.RequestUri)) };
+                }
+            }
+            return null;
+        }
+
+        private static Task<string> GetMonikerDefinition(ErrorLog errorLog, Uri url)
+        {
+            return Fetch($"{s_buildServiceEndpoint}/v2/monikertrees/allfamiliesproductsmonikers", s_opsHeaders);
```

Comment:
why no exception handling for this call since we have for the others?

Reply:
Validation service calls don't block build, when we failed to retrieve validation rule, there is a warning in the build report and validation is skipped. This behavior is the same as in Docs.Build